### ISSUE TITLE
Training loss using FedAvg and Scaffold on heart disease dataset does not converge

### ISSUE THREAD

**Issue by ybouilla** [owkin/FLamby#252]
Dear all, 

I am trying to use FLamby on heart disease dataset, but training losses does not converge, using either Scaffold or FedAvg aggregators, with all defaults hyper parameters (provided in common file) and SGD as optimizer.

Please find below the code snippet I am using (based on the quickstart session's code):
```python

import torch
from flamby.datasets.fed_heart_disease import FedHeartDisease, HeartDiseaseRaw
from flamby.datasets.fed_heart_disease import (
BATCH_SIZE,
    LR,
    metric,
    NUM_CLIENTS,
Baseline, 
BaselineLoss
)
from flamby.strategies.fed_avg import FedAvg as strat

train_dataloaders = [
            torch.utils.data.DataLoader(
                FedHeartDisease(center = i, train = True, pooled = False, normalize=True),
                batch_size = BATCH_SIZE,
                shuffle = True,
                num_workers = 0
            )
            for i in range(NUM_CLIENTS)
        ]
args = {
            "training_dataloaders": train_dataloaders,
            "model": Baseline(),
            "loss": BaselineLoss(),
            "optimizer_class": torch.optim.SGD,
            "learning_rate": LR,
            "num_updates": 60, 
            "nrounds": 20, 
            "log":True,
            "log_period": 1}

s = strat(**args)
s.run()
```
And here are the training losses I got
![image](https://user-images.githubusercontent.com/114240274/200801858-13e464a9-eb1b-44d0-afc4-7417bd13a11d.png)
![image](https://user-images.githubusercontent.com/114240274/200802127-1786284f-7b01-49ff-a4ea-5b442d157d4e.png)
![image](https://user-images.githubusercontent.com/114240274/200802240-3cb15d99-9868-47c4-b005-335602bd7fc5.png)
![image](https://user-images.githubusercontent.com/114240274/200802519-75dfab3c-87b5-4761-8da4-5b42d7244283.png)

I also tried with Scaffold but results are not getting better
Is there something I am doing wrong? 
Thank you for your time

**Comment by jeandut** [comment]
Hello @ybouilla !
The default `LR` works only for the pooled case aka when using traditional pytorch without any FL strategy, luckily we have an appendix detailing all hyper parameters that were used for FL experiments. See page 14-15 of [our arxiv submission](https://arxiv.org/abs/2210.04620). For instance for Scaffold and Fed-Heart Disease we used 0.001 (Table 16 page 14).
As those parameters were cross-validated only by looking at the final metric maybe the trainings losses will still fluctuate.
Fluctuations of the loss may come from:
1. Limited batch-size
2. Too large learning rate. 
3. 
Try using a larger batch-size and a smaller learning rate and see if the oscillations dampen.

**Comment by jeandut** [comment]
See https://github.com/owkin/FLamby/blob/main/flamby/config_heart_disease.json for the full-list of hyper-parameters used for the benchmark. We were trying to optimize the metrics given a budget in rounds:
For `num_updates=100` and `nrounds = get_nb_max_rounds(100, batch_size=BATCH_SIZE)`

**Comment by jeandut** [comment]
@ybouilla we are planning on working on benchmarks in FLamby in general using the https://benchopt.github.io/ initiative. This will allow to compare optimizers and hyperparameters more easily. Until then I am closing this issue.
